instrumentation/github.com/go-kit/kit/otelkit/propagation.go

```diff
@@ -0,0 +1,82 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package otelkit // import "go.opentelemetry.io/contrib/instrumentation/github.com/go-kit/kit/otelkit"
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/go-kit/kit/endpoint"
+	"google.golang.org/grpc/metadata"
+
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/propagation"
+)
+
+type MultimapCarrier map[string][]string
```

Comment:
This has the same underlying type as the [`HeaderCarrier`](https://github.com/open-telemetry/opentelemetry-go/blob/0f3ab76ff31af174bd0f8af54f194b91a80479f5/propagation/propagation.go#L69-L89). Why do we need to include this and not just wrap a `map[string][]string` as a `HeaderCarrier`?

Reply:
Absolutely... Thanks for pointing this out (sorry, I didn't realize `http.Header` was the same underlying).

Reply:
Ahh, it looks like a new type is needed after all - `http.Header` canonicalizes the keys passed into `Get` [^1][^2] and `metadata.MD`, which is used to store the **traceparent** header in gRPC, converts all keys to lowercase [^3][^4][^5].

The idea behind using `type MultimapCarrier map[string][]string` versus `type MetadataCarrier metadata.MD` was to relieve the dependency on `google.golang.org/grpc` package in case this type would become part of **opentelemetry-go** for other gRPC-enabled transport libraries to use, just as `HeaderCarrier` can be used for HTTP transport.

[^1]: https://pkg.go.dev/net/http#Header.Get
[^2]: https://cs.opensource.google/go/go/+/master:src/net/http/header.go;l=48 
[^3]: https://pkg.go.dev/google.golang.org/grpc/metadata#MD.Get
[^4]: https://pkg.go.dev/google.golang.org/grpc/metadata#MD.Set
[^5]: https://pkg.go.dev/google.golang.org/grpc/metadata#New